### ISSUE TITLE
Trailing whitespace

### ISSUE THREAD

**Issue by nulldatamap** [kirkedal/compSys-e2017-sim#10]
There's numerous occurrences of trailing whitespace.

Examples:
architecture_tools/sim_main.c:23
architecture_tools/sim_main.c:29
architecture_tools/sim_main.c:31
architecture_tools/sim_main.c:34
architecture_tools/sim_main.c:36
and many more.

**Comment by truls** [comment]
Yes, it appears that xcode is not very good at handling whitespace. The code is rife with trailing whitespace and mixed tab/spaces indentation

**Comment by niswegmann** [comment]
I've checked the code. And yes, I have trailing white spaces in empty lines in blocks between statements, such as in

```
{
    stmt1;
    stmt2;
    <- four trailing spaces here
    stmt3;
    stmt4;
}
```

I don't consider that an error as I use that style consistently.

Regarding the mixed spaces/tabs, I have none of those -- I searched the code for all tabs. (There are some cases of these in the annotation library, but I'm not gonna spend time on removing those.)

**Comment by truls** [comment]
Just because it's consistent, doesn't mean it's good :) It seems you can actually configure Xcode to remove trailing whitespace, including whitespace-only lines. https://stackoverflow.com/questions/1390329/trim-trailing-spaces-in-xcode (second asnwer).

And yes, your code doesn't have any mixed indentation, but @finnschiermer code does. A single indent is 4 spaces, double indent is 1 tab, tree indents is 1 tab + 4 spaces. So the code will look really strange if tabs are rendered as anything other than 8 spaces.
